src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelAPITests.cs

```diff
@@ -4041,9 +4041,15 @@ static void Main(string[] args)
             var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
             var tree = comp.SyntaxTrees.Single();
             var model = comp.GetSemanticModel(tree);
+
+            var expected = new bool[2];
+            expected[0] = true;
+            expected[1] = false;
+            var iterator = 0;
             foreach (var interp in tree.GetRoot().DescendantNodes().OfType<InterpolatedStringExpressionSyntax>())
```

Comment:
The `foreach` was useful when the whole thing was the same expected value. Now, I think it's better to just make an array of these (`tree.GetRoot().DescendantNodes().OfType<InterpolatedStringExpressionSyntax>().ToArray()`) and simply verify the expectation at each one. For the constant one, we should also verify that the constant value is what we expected.

Reply:
Changed to assert each one and check the value.

Reply:
Changed to direct asserts and a value check :+1: